admin/pre-reg/views.py

```diff
@@ -0,0 +1,171 @@
+import functools
+
+from django.contrib.auth.decorators import login_required  # , user_passes_test
+from django.views.decorators.csrf import csrf_exempt
+from django.contrib.auth.models import User
+from django.shortcuts import render
+from django.http import JsonResponse
+
+import json
+import httplib as http
+
+from modularodm import Q
+
+import utils
+
+from framework.auth.core import User as OsfUser
+from framework.mongo.utils import get_or_http_error
+from framework.exceptions import HTTPError, PermissionsError
+from website.project.model import MetaSchema, DraftRegistration
+from website.exceptions import NodeStateError
+
+get_draft_or_error = functools.partial(get_or_http_error, DraftRegistration)
+
+def load_osf_user(django_user):
+
+    if not django_user.osf_user:
+        raise RuntimeError()
+    else:
+        return OsfUser.load(django_user.osf_user.osf_id)
+
+def get_prereg_users():
```

Comment:
Django's preferred way of doing this type of thing is with a custom manager and a custom queryset: https://docs.djangoproject.com/en/1.9/topics/db/managers/#calling-custom-queryset-methods-from-the-manager

This does create quite a lot of boilerplate code so the django_model_utils package includes a PassThroughManager that encapsulates most of the boilerplate http://django-model-utils.readthedocs.org/en/latest/managers.html#passthroughmanager

The for loop with the filter in it will run one query for each user in users, plus the initial query to get users. Using the manager/queryset option should do a `MyUser.objects.filter(groups__name='prereg_group')` which will make one query and give you a list of users that are in the prereg group.

Alternatively, you could do `Group.objects.filter(name='prereg_group').user_set.all()` and that would give you all the users in the prereg_group.